scripts/sct_run_batch.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python
+##############################################################################
+#
+# Wrapper to processing scripts, which loops across subjects. Data should be
+# organized according to the BIDS structure:
+# https://github.com/sct-pipeline/spine_generic#file-structure
+#
+# ----------------------------------------------------------------------------
+# Copyright (c) 2020 Polytechnique Montreal <www.neuro.polymtl.ca>
+# Authors: Julien Cohen-Adad, Chris Hammill
+#
+# About the license: see the file LICENSE.TXT
+##############################################################################
+
+from __future__ import absolute_import
+
+import os
+import sys
+import argparse
+import numpy as np
+import multiprocessing
+import subprocess
+import re
+import time
+
+import spinalcordtoolbox.image as msct_image
+from spinalcordtoolbox.image import Image, concat_data
+from spinalcordtoolbox.utils import Metavar, SmartFormatter
+
+parser = argparse.ArgumentParser(
+  description = 'Wrapper to processing scripts, which loops across subjects. Subjects '
+                'should be organized as folders within a single directory. We recommend '
+                'following the BIDS convention (https://bids.neuroimaging.io/). '
+                'The processing script (task) should accept a subject directory as its only argument. '
+                'Additional information is passed via environment variables and the arguments '
+                'passed via `--task-args`',
+  formatter_class = SmartFormatter,
+  prog=os.path.basename(__file__).strip('.py'))
+
+parser.add_argument("--jobs", type = int, default = 1,
+                    help = 'The number of jobs to run in parallel. '
+                           'either an integer greater than or equal to one '
+                           'specifying the number of cores, 0 or a negative integer '
+                           'specifying number of cores minus that number. For example '
+                           '--jobs -1 indicates run ncores - 1 jobs in parallel.')
+parser.add_argument("--path-data", help = 'Setting for environment variable: PATH_DATA\n'
+                                          'Path containing subject directories in a consistent format')
+parser.add_argument("--subject-prefix", default = "sub-",
+                    help = 'Subject prefix, defaults to "sub-"')
+parser.add_argument("--path-output", default = "./",
+                    help = 'Base directory for environment variables:\n'
```

Comment:
trick: use "R|" at the beginning to have \n interpreted. See: [example](https://github.com/neuropoly/spinalcordtoolbox/wiki/code-style)
```suggestion
                    help = 'R|Base directory for environment variables:\n'
```

Reply:
I've fixed in the code. @jcohenadad, why does `'R|'` do the opposite of its usual behaviour here? I see in the linked document that this is the behaviour of `SmartFormatter`, but my understanding is that usually `'R|'` prevents interpretation of escape characters.

Reply:
good question! because we are using two formatters (code [here](https://github.com/neuropoly/spinalcordtoolbox/blob/master/spinalcordtoolbox/utils.py#L67)): one that adjust the text break with the terminal width, and another one that can interpret carriage return if "|R" is present. Unfortunately, both formatters cannot be used at the same time.  See example of the formatted usage from your latest commit 65429fd4e339e124dd0cce54d3fe6995f758a495:
~~~
# --> here, "|R" was not used and usage appropriately breaks lines and left-align it
  --exclude EXCLUDE     Optional regex used to filter the list of subject
                        directories. Only process a subject if they do not match the
                        regex. Exclusions are processed after inclusions.
# --> here, "|R" was used, and there is an ugly breaking of the word "task", warping it at the extreme left of the terminal. 
  --path-segmanual PATH_SEGMANUAL
                        Setting for environment variable: PATH_SEGMANUAL
                        A path containing manual segmentations to be used by the t
ask program.
~~~
If you or some else has a workaround for this, it would be awesome!

Reply:
Ahh this seems to be a tricky one. I'll give it a look and see if I can figure out a workaround

Reply:
As far as I can tell, the `_fill_text` method of `SmartFormatter` is never getting used. I tried instrumenting with a print statement with no luck. Instrumenting `_split_lines` does however get used. So it looks like `argparse.HelpFormatter` is doing most of the work. Should this be a separate issue?

Reply:
Thanks for digging! Yes if you could open an issue that would be awesome 🙏

Reply:
Conversation moved to #2685